### PR TITLE
manifest post-processor: add possibity to create hclvars files

### LINKED ISSUE #9443: [HCL2] give manifest post-processor ability to generate HCL2 varfiles
Chaining builds is awesome.

Currently the manifest post-processor generates JSON files and in HCL2 the [jsondecode](https://www.packer.io/docs/from-1.5/functions/encoding/jsondecode) function could be used to parse such a file, for example:

```hcl
locals {
  manifest_file = jsondecode(file("path/to/file.json"))
  my_output_file = manifest_file.something.something[0].output_file
}
```
Note that this could be confusing to programatically parse if a build runs two builds at a time because an example manifest file looks like:

```json
{
  "builds": [
    {
      "name": "docker",
      "builder_type": "docker",
      "build_time": 1507245986,
      "files": [
        {
          "name": "packer_example",
          "size": 102219776
        }
      ],
      "artifact_id": "Container",
      "packer_run_uuid": "6d5d3185-fa95-44e1-8775-9e64fe2e2d8f",
      "custom_data": {
        "my_custom_data": "example"
      }
    }
  ],
  "last_run_uuid": "6d5d3185-fa95-44e1-8775-9e64fe2e2d8f"
}
```
And we cannot guarantee an order here.

Another way to do this could be to allow generating hcl2 varfiles, for example:

```hcl
build {
  # ...

  post-processor "manifest" {
    # because the file is a '.pkrvars.hcl' file hcl version will be used.
    output = "manifest-${source.type}-${source.name}.pkrvars.hcl" 
    var_name = "my_docker_build"
  }
}
```

This could generate something like:
```hcl
#manifest.docker.my-linux.pkrvars.hcl
my_docker_build = {
    name = "docker"
    builder_type = "docker"
    build_time = 1507245986
    files = [
    {
        "name": "packer_example",
        "size": 102219776,
    },
    ]
    artifact_id = "Container"
    packer_run_uuid = "6d5d3185-fa95-44e1-8775-9e64fe2e2d8f"
    custom_data = {
        "my_custom_data": "example"
    }
}
```

That file could be used using the `-var-file` option etc.

### PR DESCRIPTION
**Pull request by azr**
close #9443 

This will allow to reuse values from a previous build.

* tests
* docs

more details at the bottom of website/pages/docs/post-processors/manifest.mdx

### CODE REVIEW

File: website/pages/docs/post-processors/manifest.mdx [hunk 1]
```diff
@@ -202,3 +227,20 @@ AMI_ID=$(jq -r '.builds[-1].artifact_id' manifest.json | cut -d ":" -f2)
...
+chain builds.
+
+For example
+
+```hcl
+source "amazon-ebs" "chained" {
+  source_ami = var.manifest.amazon-ebs.old_build.artifact_id
+}
+```
+
```
Comment (azr) [inline website/pages/docs/post-processors/manifest.mdx:244]: Is this enough of an explanation ?

File: website/pages/docs/post-processors/manifest.mdx [hunk 2]
```diff
@@ -202,3 +227,35 @@ AMI_ID=$(jq -r '.builds[-1].artifact_id' manifest.json | cut -d ":" -f2)
...
+```hcl
+variable "manifest" {
+  type = any
+}
+
+source "amazon-ebs" "chained" {
+  source_ami = var.manifest.amazon-ebs.old_build.artifact_id
+}
+```
```
Comment (azr) [inline website/pages/docs/post-processors/manifest.mdx:258]: The explanation is here; I hope it's good enough 🤔

**Comment by SwampDragons** [comment]
With json manifests, the manifest gets appended to with each run rather than overwriting the old manifest. This behavior is different. Is it not possible with hcl?

**Comment by azr** [comment]
Ah, you are right ! I have over simplified it 🤔 
With this behaviour a similarly named manifest is overwritten on each run, (new but different builds would be appended). I'm building a new full example to refresh my memory:

```hcl
manifest = {
  null = {
    second-example = {
      # when 'second-example' is rebuilt this will be overwritten
      artifact_id     = "Null"
```

now, one can access :

```hcl
 x = maniferst.null.second-example.artifact_id
```

But if we generated:

```hcl
manifest = {
  last_run_uuid = "uuid-something-blah-888"
  runs {
    uuid-something-blah-888 {
      null = {
        second-example = {
          artifact_id     = "Null"
```

We could probably do :

```hcl
  x = manifest.runs[manifest.last_run_uuid].null.second-example.artifact_id
```

Also if we generated:

```hcl
manifest = {
  runs [
    null = {
      second-example = {
        artifact_id     = "Null"
```

We could probably do :

```hcl
  x = manifest.runs[0].null.second-example.artifact_id
```

I need to think a little more about this 🤔

**Comment by puetzk** [comment]
Would be nice if this also had a way to omit any [sensitive-variables](https://www.packer.io/docs/templates/user-variables.html#sensitive-variables) - it would be nice in complex pipelines for CI to report all the settings it used, but if you want to locally repeat a piece you'll have to bring your own passwords...

**Comment by azr** [comment]
Converting to draft as it might become different

**Review by park-cheif5 — COMMENTED** [review]
Test and see how it goes id like to see it simplified to load less.

**Comment by park-cheif5** [comment]
Perfect

**Comment by azr** [comment]
closing this one for now ! Hang in there !

**Pull request closed without merging**